src/Integrations/Integrations/Swoole/SwooleIntegration.php

```diff
@@ -0,0 +1,202 @@
+<?php
+
+namespace DDTrace\Integrations\Swoole;
+
+use DDTrace\HookData;
+use DDTrace\Integrations\Integration;
+use DDTrace\SpanData;
+use DDTrace\Tag;
+use DDTrace\Type;
+use DDTrace\Util\Normalizer;
+use Swoole\Http\Request;
+use Swoole\Http\Response;
+use Swoole\Http\Server;
+use function DDTrace\consume_distributed_tracing_headers;
+use function DDTrace\extract_ip_from_headers;
+
+class SwooleIntegration extends Integration
+{
+    const NAME = 'swoole';
+
+    public function getName()
+    {
+        return self::NAME;
+    }
+
+    public function addTraceAnalyticsIfEnabled(SpanData $span)
+    {
+        if (!$this->configuration->isTraceAnalyticsEnabled()) {
+            return;
+        }
+        $span->metrics[Tag::ANALYTICS_KEY] = $this->configuration->getTraceAnalyticsSampleRate();
+    }
```

Comment:
Why is this here? Isn't that inherited from the Integration class?

Reply:
I suppose you wanted to copy `requiresExplicitTraceAnalyticsEnabling`, which should indeed be false here.

Reply:
Yes